flycheck.el

```diff
@@ -5596,6 +5597,30 @@ See URL `http://phpmd.org/' for more information about phpmd."
                       errors)))))))))
        (nreverse errors)))))
 
+(defun flycheck-parse-reek (output checker buffer)
+  "Parse Reek warnings from JSON OUTPUT.
+
+CHECKER and BUFFER denoted the CHECKER that returned OUTPUT and
+the BUFFER that was checked respectively.
+
+See URL `https://github.com/troessner/reek' for more information
+about Reek."
+  (let ((errors))
+    (dolist (message (car (flycheck-parse-json output)))
+      (let-alist message
+        (dolist (line (delete-dups .lines))
+          (push
+           (flycheck-error-new-at
+            line
+            nil
+            'warning (concat .context " " .message)
```

Comment:
I was thinking about that, but decided not to do it. The reasons are:
- This way warning is normal sentence
- When requesting standard text output reek does it same way:

```
  [15, 15, 16]:DuplicateMethodCall: ApplicationController#validate_type calls 'params['data']' 3 times [https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md]
```